SQL/New_patches/2018-01-17-normalisation_visit.sql

```diff
@@ -1,25 +1,27 @@
+ALTER TABLE project_rel DROP PRIMARY KEY, ADD COLUMN `ProjectSubprojectRelID` int(10) unsigned NOT NULL PRIMARY KEY AUTO_INCREMENT FIRST;
```

Comment:
some version of MySQL will complain.
```suggestion
ALTER TABLE project_rel DROP PRIMARY KEY;
ALTER TABLE project_rek ADD COLUMN `ProjectSubprojectRelID` int(10) unsigned NOT NULL PRIMARY KEY AUTO_INCREMENT FIRST;
```

Reply:
your suggestion gives me an error on mariadb.

Reply:
What's the error? The deleted line (trying to drop and add in the same command) looks weird to me.

Reply:
```
Query execution failed

Reason:
SQL Error [1075] [42000]: (conn:228) Incorrect table definition; there can be only one auto column and it must be defined as a key
Query is : ALTER TABLE project_rel DROP PRIMARY KEY
```

Reply:
The error is not with the DROP statement, it is in the order of the parameter of the ADD COLUMN.  
`PRIMARY KEY` as to be after the `AUTO_INCREMENT`.

I haven't check well the ordering in your instruction.

Reply:
but if you see the error message. I get the error when I run the `ALTER TABLE project_rel DROP PRIMARY KEY` query

Reply:
Where you trying the entire drop and add in one statement?

Could you post the exact query and result here?

on my VM:
```
mysql> ALTER TABLE project_rel DROP PRIMARY KEY;
Query OK, 6 rows affected (0.46 sec)
Records: 6  Duplicates: 0  Warnings: 0
mysql> ALTER TABLE project_rel ADD COLUMN `ProjectSubprojectRelID` int(10) unsigned NOT NULL  AUTO_INCREMENT PRIMARY KEY FIRST;
Query OK, 0 rows affected (0.10 sec)
Records: 0  Duplicates: 0  Warnings: 0
```

Reply:
```
mysql> ALTER TABLE project_rel DROP PRIMARY KEY;
SQL Error [1075] [42000]: (conn:228) Incorrect table definition; there can be only one auto column and it must be defined as a key
```

Reply:
Can you post the current definition of you table?
```
SHOW CREATE TABLE project_rel
```

Reply:
NEVERMINDDDDD MA BAD